lektor/publisher.py

```diff
@@ -657,6 +657,9 @@ def git(args, **kwargs):
 def publish(env, target, output_path, credentials=None, **extra):
     url = urls.url_parse(text_type(target))
     publisher = env.publishers.get(url.scheme)
+    if url.host is None:
+        raise PublishError('missing host name')
```

Comment:
Can you provide a little more context in this error message? When writing error messages, consider what information you'd need to know in order to fix the problem. If I saw "missing host name" in a traceback, my first question would be, "_What_ is missing a host name?"

I would recommend something like this:
```python
raise PublishError("Publishing target does not have a host name: %s", target)
```